### PR TITLE
OBMIN-433 create new endpoint for add favorite

### PR DESCRIPTION
**Pull request by ImAndVlad**
- Create new endpoint for add favorite
- Added tests

### CODE REVIEW

**Review by IllyaZuew — APPROVED** [review]

File: src/test/java/space/obminyashka/items_exchange/service/AdvertisementServiceTest.java [hunk 1]
```diff
@@ -49,6 +49,15 @@ void isAdvertisementsHaveSubcategoryWithId_whenTryToCheckSubcategory_shouldSameR
         verify(advertisementRepository).existsBySubcategoryId(NONEXISTENT_ENTITY_ID);
     }
 
+    @Test
+    void addFavorite_WhenDataCorrect_Successfully() {
+        final var existedAdvId = UUID.randomUUID();
+        final var existedUsername = "user";
```
Comment (IllyaZuew) [inline src/test/java/space/obminyashka/items_exchange/service/AdvertisementServiceTest.java:55]: Add enter here. You need to show 3 section given when taken

File: src/test/java/space/obminyashka/items_exchange/end2end/UserFlowTest.java [hunk 2]
```diff
@@ -141,6 +141,17 @@ void getFavoriteAdvertisements_shouldReturnPage_WhenHaveSpecialPageAndSize() thr
                 .andExpect(jsonPath("$.content[0].title").value("Blouses"));
     }
 
+    @Test
+    @WithMockUser(username = ADMIN_USERNAME)
+    @DataSet("database_init.yml")
```
Comment (IllyaZuew) [inline src/test/java/space/obminyashka/items_exchange/end2end/UserFlowTest.java:146]: add here @ExpectedDataSet

**Review by IllyaZuew — APPROVED** [review]

File: src/main/java/space/obminyashka/items_exchange/controller/UserController.java [hunk 3]
```diff
@@ -105,6 +106,24 @@ public Page<AdvertisementTitleDto> getFavoriteAdvertisements(
...
+            @ApiResponse(responseCode = "202", description = "ACCEPTED"),
+            @ApiResponse(responseCode = "401", description = "UNAUTHORIZED"),
+            @ApiResponse(responseCode = "404", description = "NOT_FOUND")})
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    public void addFavoriteAdvertisement(
+            @Parameter(hidden = true) Authentication authentication,
+            @Parameter(name = "advertisementId", description = "Id of advertisement for to add an favorite adv")
+            @PathVariable UUID advertisementId) {
+        if (!advService.existById(advertisementId)) {
+            throw new EntityIdNotFoundException(getMessageSource(ResponseMessagesHandler.ExceptionMessage.ADVERTISEMENT_NOT_EXISTED_ID));
```
Comment (rpkyrych) [inline src/main/java/space/obminyashka/items_exchange/controller/UserController.java:122]: Would you mind applying here static import for the exception message key?

File: src/main/java/space/obminyashka/items_exchange/api/ApiKey.java [hunk 4]
```diff
@@ -48,6 +48,7 @@ public class ApiKey {
     public static final String USER_MY_INFO = USER + "/my-info";
     public static final String USER_MY_ADV = USER + "/my-adv";
     public static final String USER_MY_FAVORITE = USER_MY_ADV + "/favorite";
+    public static final String USER_ADD_MY_FAVORITE = USER_MY_FAVORITE + "/{advertisementId}";
     public static final String USER_DELETE_MY_FAVORITE = USER_MY_FAVORITE + "/{advertisementId}";
```
[the remark this answers is not in the document]
Comment (IllyaZuew) [reply src/main/java/space/obminyashka/items_exchange/api/ApiKey.java:52]: Open Closed principle?)

File: src/main/java/space/obminyashka/items_exchange/controller/UserController.java (hunk 3 above)
Comment (ImAndVlad) [reply src/main/java/space/obminyashka/items_exchange/controller/UserController.java]: made

**Review by IllyaZuew — DISMISSED** [review]

**Review by ennnbaev — DISMISSED** [review]

**Review by demus-n — DISMISSED** [review]

**Review by Sershok — DISMISSED** [review]

File: src/test/java/space/obminyashka/items_exchange/end2end/UserFlowTest.java [hunk 5]
```diff
@@ -141,19 +141,31 @@ void getFavoriteAdvertisements_shouldReturnPage_WhenHaveSpecialPageAndSize() thr
...
+    @ExpectedDataSet(value = "advertisement/addFavorite.yml")
+    void addFavoriteAdvertisement_shouldReturnException_whenAdvertisementIsNotFound() throws Exception {
```
Comment (rpkyrych) [inline src/test/java/space/obminyashka/items_exchange/end2end/UserFlowTest.java:148]: Do we have any positive end2end test for that logic?

**Review by IllyaZuew — APPROVED** [review]

File: src/test/java/space/obminyashka/items_exchange/end2end/UserFlowTest.java (hunk 5 above)
Comment (ImAndVlad) [reply src/test/java/space/obminyashka/items_exchange/end2end/UserFlowTest.java:157]: +

**Review by Sershok — APPROVED** [review]

**Review by ennnbaev — APPROVED** [review]

**Review by rpkyrych — APPROVED** [review]

**Pull request merged**